notebooks/working/3.0-mab-independent-variable-simulation.md

```diff
@@ -0,0 +1,1127 @@
+#         utility specification the explanatory variable is entering.
+
+mnl_specification = OrderedDict()
+mnl_names = OrderedDict()
+
+mnl_specification["intercept"] = [2, 3, 4, 5, 6, 7, 8]
+mnl_names["intercept"] = ['ASC Shared Ride: 2',
+                          'ASC Shared Ride: 3+',
+                          'ASC Walk-Transit-Walk',
+                          'ASC Drive-Transit-Walk',
+                          'ASC Walk-Transit-Drive',
+                          'ASC Walk',
+                          'ASC Bike']
+
+mnl_specification["total_travel_time"] = [[1, 2, 3], [4, 5, 6]]
+mnl_names["total_travel_time"] = ['Travel Time, units:min (All Auto Modes)',
+                                  'Travel Time, units:min (All Transit Modes)']
+
+mnl_specification["total_travel_cost"] = [[4, 5, 6]]
+mnl_names["total_travel_cost"] = ['Travel Cost, units:$ (All Transit Modes)']
+
+mnl_specification["cost_per_distance"] = [1, 2, 3]
+mnl_names["cost_per_distance"] = ["Travel Cost per Distance, units:$/mi (Drive Alone)",
+                                  "Travel Cost per Distance, units:$/mi (SharedRide-2)",
+                                  "Travel Cost per Distance, units:$/mi (SharedRide-3+)"]
+
+mnl_specification["cars_per_licensed_drivers"] = [[1, 2, 3]]
+mnl_names["cars_per_licensed_drivers"] = ["Autos per licensed drivers (All Auto Modes)"]
+
+mnl_specification["total_travel_distance"] = [7, 8]
+mnl_names["total_travel_distance"] = ['Travel Distance, units:mi (Walk)',
+                                      'Travel Distance, units:mi (Bike)']
+
+# mnl_specification["cross_bay"] = [[2, 3], [4, 5, 6]]
+# mnl_names["cross_bay"] = ["Cross-Bay Tour (Shared Ride 2 & 3+)",
+#                           "Cross-Bay Tour (All Transit Modes)"]
+mnl_specification["cross_bay"] = [[2, 3]]
+mnl_names["cross_bay"] = ["Cross-Bay Tour (Shared Ride 2 & 3+)"]
+
+mnl_specification["household_size"] = [[2, 3]]
+mnl_names["household_size"] = ['Household Size (Shared Ride 2 & 3+)']
+
+mnl_specification["num_kids"] = [[2, 3]]
+mnl_names["num_kids"] = ["Number of Kids in Household (Shared Ride 2 & 3+)"]
+```
+
+```python
+# Estimate the basic MNL model, using the hessian and newton-conjugate gradient
+mnl_model = pl.create_choice_model(data=bike_data_long,
+                                   alt_id_col="mode_id",
+                                   obs_id_col="observation_id",
+                                   choice_col="choice",
+                                   specification=mnl_specification,
+                                   model_type="MNL",
+                                   names=mnl_names)
+
+num_vars = len(reduce(lambda x, y: x + y, mnl_names.values()))
+# Note newton-cg used to ensure convergence to a point where gradient
+# is essentially zero for all dimensions.
+mnl_model.fit_mle(np.zeros(num_vars),
+                  method="BFGS")
+
+# Look at the estimation results
+mnl_model.get_statsmodels_summary()
+```
+
+# Simulate Data - Based on Wide Data
```

Comment:
Any reason this section exits still, other than historical reasons?

Just wanted to make sure I'm not missing anything!

Reply:
No reason at all, just for tracking the progression. I can delete after today's mtg.